src/Connect.js

```diff
@@ -470,6 +484,31 @@ class Connect {
   genCallback(reqId) {
     return this.isOnMobile ?  windowCallback(reqId) : transport.messageServer.genCallback()
   }
+
+  /**
+   * @private
+   * Sign a profile object with this.credentials, and upload it to ipfs, prepending
+   * the instance array of verified claims (this.vc) with the ipfs uri.  If a profile
+   * object is not provided, create one on the fly 
+   * @param {Object}  [profile]         the profile object to be signed and uploaded
+   * @returns {Promise<String, Error>}  a promise resolving to the ipfs hash, or rejecting with an error
+   */
+  signAndUploadProfile(profile) {
+    if (!profile && this.vc.length > 0) return
+    profile = profile || {
+      name: this.appName,
+      description: this.description,
+      url: (typeof window !== 'undefined') ? window.location.host : undefined, 
+    }
+    
+    // Upload to ipfs
+    return this.credentials.signJWT(profile)
```

Comment:
I think this should be `createVerification({sub:this.did, claims:profile})` or something like that

Reply:
Actually it really should really be a verification for each attribute.